tools/python/google_maps_bookmarks.py

```diff
@@ -104,26 +140,51 @@ def write_kml(self):
             point = ET.SubElement(placemark, "Point")
             ET.SubElement(point, "coordinates").text = place['coordinates']
         tree = ET.ElementTree(root)
-        tree.write(self.kml_file)
-        print()
+        tree.write(self.output_file)
         print()
-        print("Exported Google Saved Places to " + os.path.abspath(self.kml_file))
-
-
-def get_json(url):
-    max_attempts = 3
-    for retry in range(max_attempts):
-        try:
-            response = urllib.request.urlopen(url)
-            return json.load(response)
-        except urllib.error.URLError:
-            print(f"Couldn't connect to Google Maps. Retrying... ({retry + 1}/{max_attempts})")
-            if retry < max_attempts - 1:
-                continue
+        print("Exported Google Saved Places to " + path.abspath(self.output_file))
+
+    def write_gpx(self):
+        gpx = ET.Element("gpx", version="1.1", creator="GoogleMapsConverter")
+        for place in self.places:
+            wpt = ET.SubElement(gpx, "wpt", lat=place['coordinates'].split(',')[1], lon=place['coordinates'].split(',')[0])
+            ET.SubElement(wpt, "name").text = place['name']
+            ET.SubElement(wpt, "desc").text = place['description']
+        tree = ET.ElementTree(gpx)
+        tree.write(self.output_file)
+        print("Exported Google Saved Places to " + path.abspath(self.output_file))
+
+    def convert(self):
+        with open(self.input_file, 'r') as file:
+            content = file.read().strip()
+            if not content:
+                raise ValueError(f"The file {self.input_file} is empty or not a valid JSON file.")
+            
+            # Determine the file mime type (GeoJSON or CSV) and process accordingly
+            mime_type, _ = mimetypes.guess_type(self.input_file)
+            if mime_type == 'application/geo+json' or mime_type == 'application/json':
+                try:
+                    geojson = json.loads(content)
+                except json.JSONDecodeError:
+                    raise ValueError(f"The file {self.input_file} is not a valid JSON file.")
+                self.process_geojson_features(geojson)
+            elif mime_type == 'text/csv':
+                self.get_api_key()
+                self.process_csv_features(content)
             else:
-                raise
+                raise ValueError(f"Unsupported file format: {self.input_file}")
+        
+        # Write to output file in the desired format, KML or GPX
+        if self.output_format == 'kml':
+            self.write_kml()
+        elif self.output_format == 'gpx':
+            self.write_gpx()
 
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Convert Google Maps saved places to KML or GPX.")
```

Comment:
The script should either be called with arguments or interactively. You could add `input` and `format` to the interactive input or set the bookmarks name and API key with arguments.